pageserver/src/tenant/storage_layer/layer.rs

```diff
@@ -675,6 +679,8 @@ impl LayerInner {
                     .upgrade()
                     .ok_or_else(|| DownloadError::TimelineShutdown)?;
 
+                // FIXME: grab a gate
```

Comment:
For awareness: last week I spent a couple of hours playing with this: https://github.com/neondatabase/neon/pull/5967

Reply:
Left comments.